### PR TITLE
[FIX] Changing env to mountable path - PRIVATE_REGISTRY_CREDENTIAL_PATH

### PR DESCRIPTION
**Pull request by harini-venkataraman**
## What
Initally the credentials for authenticating private images were directly set to the env variable.
This PR changes to a mountable path.
...

## Why
This change helps automating On-prem deployents
...

## How
Introducing PRIVATE_REGISTRY_CREDENTIAL_PATH for uploading the mount path.

...

## Can this PR break any existing features. If yes please list of possible items. If no please exaplin why. (PS: Admins do not merge the PR without this section filled)

This PR will not break existing features.
It introduces an env, where the path of SA Key file is mounted.
If the file is not present, but the env is set - We do not raise exception rather catch and log.
This would prevent the constructor from failing in cases where the private tool is not used. 

...

## Database Migrations
Not applicable

...

## Env Config
PRIVATE_REGISTRY_CREDENTIAL_PATH - Path  where the credential file is mounted.

...

## Relevant Docs
Not applicable

...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: worker/src/unstract/worker/worker.py [hunk 1]
```diff
@@ -26,19 +26,29 @@ def __init__(self, image_name: str, image_tag: str) -> None:
...
-                password=private_registry_password,
-                registry=private_registry_url,
-            )
+            try:
+                open(private_registry_credential_path)
+                self.client.login(
+                    username=private_registry_username,
+                    password=open(private_registry_credential_path).read(),
+                    registry=private_registry_url,
+                )
```
Comment (gaya3-zipstack) [inline worker/src/unstract/worker/worker.py:45]: Don't we have to handle auth exception variants also for self.client.login()?

File: worker/src/unstract/worker/worker.py [hunk 2]
```diff
@@ -26,19 +26,29 @@ def __init__(self, image_name: str, image_tag: str) -> None:
...
-            self.client.login(
-                username=private_registry_username,
-                password=private_registry_password,
-                registry=private_registry_url,
-            )
+            try:
+                open(private_registry_credential_path)
+                self.client.login(
+                    username=private_registry_username,
+                    password=open(private_registry_credential_path).read(),
```
[the remark this answers is not in the document]
Comment (harini-venkataraman) [reply worker/src/unstract/worker/worker.py]: Addressed.

File: worker/src/unstract/worker/worker.py (hunk 1 above)
Comment (harini-venkataraman) [reply worker/src/unstract/worker/worker.py:47]: Addressed.

**Comment by harini-venkataraman** [comment]
> Also, do we need any docker-compose file changes for mounting the file?

@ritwik-g You have any comments to add here ?

**Review by gaya3-zipstack — APPROVED** [review]
Looks good

**Comment by gaya3-zipstack** [comment]
@harini-venkataraman @nehabagdia I have approved the PR. However I and Harini discussed a few points in propagating this error to the end user in a readable and meaningful way. @harini-venkataraman mentioned that she will address it once @muhammad-ali-e 's [work](https://github.com/Zipstack/unstract/pull/302) gets merged.

**Comment by harini-venkataraman** [comment]
> @harini-venkataraman @nehabagdia I have approved the PR. However I and Harini discussed a few points in propagating this error to the end user in a readable and meaningful way. @harini-venkataraman mentioned that she will address it once @muhammad-ali-e 's [work](https://github.com/Zipstack/unstract/pull/302) gets merged.

@gaya3-zipstack @nehabagdia Have raised a issue to track the same
https://zipstack.atlassian.net/browse/UN-1214

**Pull request merged**